governance/program/src/processor/process_create_account_governance.rs

```diff
@@ -0,0 +1,52 @@
+//! Program state processor
+
+use crate::{
+    state::account_governance::AccountGovernance,
+    state::{
+        account_governance::{get_account_governance_address_seeds, GovernanceConfig},
+        enums::GovernanceAccountType,
+    },
+    tools::account::create_and_serialize_account_signed,
+};
+use solana_program::{
+    account_info::{next_account_info, AccountInfo},
+    entrypoint::ProgramResult,
+    pubkey::Pubkey,
+    rent::Rent,
+    sysvar::Sysvar,
+};
+
+/// Processes CreateAccountGovernance instruction
+#[allow(clippy::too_many_arguments)]
+pub fn process_create_account_governance(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    config: GovernanceConfig,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+
+    let account_governance_info = next_account_info(account_info_iter)?; // 0
+    let payer_info = next_account_info(account_info_iter)?; // 1
+    let system_info = next_account_info(account_info_iter)?; // 2
+
+    let rent_sysvar_info = next_account_info(account_info_iter)?; // 3
+    let rent = &Rent::from_account_info(rent_sysvar_info)?;
+
+    let account_governance_data = AccountGovernance {
+        account_type: GovernanceAccountType::AccountGovernance,
+        config: config.clone(),
```

Comment:
This seems prone to be accidentally misused, since it just accepts whatever realm is specified in the config.  We should probably also provide the realm account, and verify it's actually a realm.

Also, I may be missing it, but are there any checks on the other fields of the config? Mainly, are we sure that all of the `u8`s are real percentages?  And probably that the `voting threshold >= proposal creation threshold`.  Any others?

I'm not sure if we also want to validate the waiting times are within a set of bounds.

And with that, you'll want to add some tests to fail correctly if the realm doesn't exist.

Reply:
Added assert_is_valid_realm and relevant tests and now working on the other checks

Reply:
Config validation added 
- `Realm` is checked to be a valid account (initialised, correct type and owner)
- `vote_threshold_percentage` checked to be in <50,100> range
- I don't think if there are any bounds for the time values. In theory it could be anything, so no validation
- I replaced `token_threshold_percentage_to_create_proposal`with `min_tokens_to_create_proposal`. Percentage was not a good idea because if there are more tokens minted we might need bps or even smaller units to accept users with a single token for example. 
- It also made me thinking about transferring upgrade_authority at `Governance` creation time. Since it's a big commitment and any mistakes might be irreversible I made it optional. If the authority is not transferred then program checks the current `upgrade_authority` signed the transaction to prevent others from creating governances for programs they don't own. The authority then can be transfer to Governance at alter time in this case.